src/vscodeUtils.ts

```diff
@@ -1,4 +1,42 @@
 import * as vscode from 'vscode';
+import * as path from 'path';
+
+export class ServerPath {
+	private folderName = 'lsp';
+	private customBinPath: string;
+	public customBinPathOptionName = 'languageServer.pathToBinary';
```

Comment:
I suspect this is not going to change dynamically at runtime any time soon?
I think it would be more common to put this as
`const CUSTOM_BIN_PATH_OPTION_NAME = 'languageServer.pathToBinary';`
above the `export class ServerPath {` line 🙂

Reply:
Same probably for `folderName`.

Reply:
Yeah, that's fair - I had the tendency to do it actually. The only reason I didn't is because constants seem to be scoped to the whole file in TypeScript, not a class AFAICT?

I could extract the class into its own file (`serverPath.ts`) I suppose - I just don't like the idea of a seemingly random constant lingering in a file called "utils" 😂 😭  but I haven't had the time yet to break down the logic here and get rid of "utils".

Reply:
Ah, I see that reasoning 👍 You could also add a `readonly` modifier and change it to a `static` to it to make it more of a constant value.
```suggestion
	static readonly customBinPathOptionName = 'languageServer.pathToBinary';
```
I think you'd also have to adjust the places it's used and replace `this` with the name of the class then.

Reply:
I extracted it into its own file. 👍🏻